pkg/controllers/nodeclass/status/validation.go

```diff
@@ -40,6 +49,32 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
 			fmt.Sprintf("%q tag does not pass tag validation requirements", offendingTag))
 		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("%q tag does not pass tag validation requirements", offendingTag))
 	}
+	//Auth Validation
+	//validates createfleet, describelaunchtemplate, createtags, and terminateinstances
+	//nolint:ineffassign, staticcheck
+	ctx = context.WithValue(ctx, "DryRun", lo.ToPtr(true))
+	if nodeClass.StatusConditions().Get(v1.ConditionTypeSubnetsReady).IsFalse() || nodeClass.StatusConditions().Get(v1.ConditionTypeAMIsReady).IsFalse() {
+		return reconcile.Result{}, nil
+	}
+	nodeClaim := coretest.NodeClaim()
+	nodeClaim.Spec.NodeClassRef.Name = nodeClass.Name
+	var errs []error
+	//create checks createfleet, and describelaunchtemplate
+	if _, err := n.cloudProvider.Create(ctx, nodeClaim); err != nil {
+		errs = append(errs, fmt.Errorf("create: %w", err))
+	}
+
+	if err := n.instanceProvider.Delete(ctx, "mock-id"); err != nil {
```

Comment:
Why are we checking delete when it comes to determining whether we can launch an instance?